src/main/src/main/java/org/geoserver/data/jdbc/SpringUnWrapper.java

```diff
@@ -30,140 +14,140 @@
-        Connection unwrapped = unwrapInternal(conn);
-        if (unwrapped != null) return unwrapped;
-        else
-            throw new IllegalArgumentException(
-                    "This connection is not unwrappable, "
-                            + "check canUnwrap before calling unwrap");
-    }
-
-    private Connection unwrapInternal(Connection conn) {
-        for (int i = 0; i < EXTRACTORS.size(); i++) {
-            NativeJdbcExtractor extractor = EXTRACTORS.get(i);
-            try {
-                // the contract is that the original connection is returned
-                // if unwrapping was not possible
-                Connection unwrapped = extractor.getNativeConnection(conn);
-
-                if (conn != unwrapped) {
-                    if (i != 0) {
-                        // move the extractor to the top, so that we don't do
-                        // many useless attempts at unwrapping with the others
-                        // (this code is typically executed for each feature)
-                        EXTRACTORS.add(0, extractor);
-                        EXTRACTORS.remove(i);
-                    }
-                    return unwrapped;
-                }
-            } catch (Throwable t) {
-                // catch a throwable since some of the unwrappers do not blow up
-                // during initialization when the enviroment does not help, but
-                // they do at unwrap time and they throw Error suclasses
-                // We just want to skip the unwrapper and move on
-            }
-        }
-        return null;
-    }
-
-    public boolean canUnwrap(Statement st) {
-        Statement unwrapped = unwrapInternal(st);
-        return unwrapped != null;
-    }
-
-    public Statement unwrap(Statement statement) {
-        Statement unwrapped = unwrapInternal(statement);
-        if (unwrapped != null) return unwrapped;
-        else
-            throw new IllegalArgumentException(
-                    "This statement is not unwrappable, "
-                            + "check canUnwrap before calling unwrap");
-    }
-
-    private Statement unwrapInternal(Statement st) {
-        for (int i = 0; i < EXTRACTORS.size(); i++) {
-            NativeJdbcExtractor extractor = EXTRACTORS.get(i);
-            try {
-                // the contract is that the original connection is returned
-                // if unwrapping was not possible
-                Statement unwrapped = extractor.getNativeStatement(st);
-                if (st != unwrapped) {
-                    if (i != 0) {
-                        // move the extractor to the beginning, so that we don't do
-                        // many useless attempts at unwrapping with the others
-                        // (this code is typically executed for each feature)
-                        EXTRACTORS.add(0, extractor);
-                        EXTRACTORS.remove(i);
-                    }
+public class SpringUnWrapper {
```

Comment:
This class should just be removed, no reason to keep it empty. The SPI registration should also be removed:
https://github.com/geoserver/geoserver/blob/0920b0a418486de4d68aa8261f937f8e99217a54/src/main/src/main/resources/META-INF/services/org.geotools.data.jdbc.datasource.UnWrapper#L2